libvast/include/vast/detail/algorithms.hpp

```diff
@@ -8,11 +8,26 @@
 
 #pragma once
 
+#include "vast/concepts.hpp"
+
 #include <algorithm>
 #include <vector>
 
 namespace vast::detail {
 
+template <class T>
+concept has_contains = requires(T& t, typename T::value_type& x) {
+  requires concepts::container<T>;
+  requires t.contains(x);
```

Comment:
```suggestion
  { t.contains(x) } -> std::convertible_to<bool>;
```

I don't think you're checking what you want to check otherwise.